packages/@dcl/inspector/src/components/Tree/Tree.tsx

```diff
@@ -78,11 +82,11 @@ function Tree<T>(_props: Props<T>) {
       [getId, getChildren]
     )
 
-    const [, drag] = useDrag(() => ({ type: 'tree', item: { value } }), [value])
+    const [, drag] = useDrag(() => ({ type: dndType, item: { value, transform: getDragValue(value) } }), [value])
```

Comment:
mm okay this now does not break the drop handler, but IMO this seems too specific for this use case. I understand that with the current implementation just having `value` is not enough and that you want to be able to get the full `TreeNode`.

I would suggest the following to address the problem of passing `tree` from the context of the `ProjectView` into the one of the `GltfInspector`:

1) In `Tree.tsx`:

Replace line 31
```ts
getDragValue?: (value: T) => any
```` 
with 
```ts
getDragContext?: () => any
```

And then instead of this on line 85:

```ts
transform: getDragValue(value)
```
do this:
```ts
context: getDragContext && getDragContext()
```

Now we have a generic way off passing data between the source (ProjectView) and the target (GltfInspector) that is not constrained to only transforming the `value`.

2) In `ProjectView.tsx`:

Replace 
```ts
getDragValue={(val) => tree.get(val)!}
```
with
```ts
getDragContext={() => ({ tree })}
```

We pass the whole tree within the context

3) Finally on `GltfInspector.tsx`:

Replace this:

```ts
 drop: ({ transform }: { transform: AssetNodeItem }, monitor) => {
  if (monitor.didDrop()) return
  handleDrop(transform)
}
```
With this
```ts
drop: ({ value, context }: { value: string; context: { tree: Map<string, AssetNodeItem> } }, monitor) => {
  if (monitor.didDrop()) return
  handleDrop(context.tree.get(value)!)
},
```

It's a little bit more code, but I believe it's more generic and could be useful for any use case that requires passing stuff from source to target that is different to `T`

Reply:
sound good. Instead of providing a way to change the value before dragging, we provide context to change it after dropping 👍